src/analysis/bounded_variation.lean

```diff
@@ -634,6 +635,147 @@ lemma monotone_on.has_locally_bounded_variation_on {f : α → ℝ} {s : set α}
   has_locally_bounded_variation_on f s :=
 λ a b as bs, ((hf.evariation_on_le as bs).trans_lt ennreal.of_real_lt_top).ne
 
+section variation_on_from_to
+
+
+/--
+The **signed** variation of `f` on the interval `Icc a b` intersected with the set `s`,
+squashed to a real (therefore only really meaningful if the variation is finite)
+-/
+noncomputable def variation_on_from_to (f : α → E) (s : set α) (a b : α) : real :=
+if a ≤ b then (evariation_on f (s ∩ Icc a b)).to_real else
+            - (evariation_on f (s ∩ Icc b a)).to_real
+
+variables {f : α → E} {s : set α}
+
+lemma variation_on_from_to_self (a : α) : variation_on_from_to f s a a = 0 :=
+begin
+  dsimp only [variation_on_from_to],
+  rw [if_pos (le_refl _), Icc_self, evariation_on.subsingleton, ennreal.zero_to_real],
+  exact λ x hx y hy, hx.2.trans hy.2.symm,
+end
+
+lemma variation_on_from_to_nonneg_of_le {a b : α} (h : a ≤ b) : 0 ≤ variation_on_from_to f s a b :=
+by { dsimp only [variation_on_from_to], simp only [if_pos h, ennreal.to_real_nonneg], }
+
+lemma variation_on_from_to_nonpos_of_ge {a b : α} (h : b ≤ a) : variation_on_from_to f s a b ≤ 0 :=
+begin
+  dsimp only [variation_on_from_to],
+  split_ifs with h',
+  { cases le_antisymm h h',
+    rw [Icc_self, evariation_on.subsingleton, ennreal.zero_to_real],
+    exact λ x hx y hy, hx.2.trans hy.2.symm, },
+  { simp only [right.neg_nonpos_iff, ennreal.to_real_nonneg], },
+end
+
+lemma variation_on_from_to_eq_of_le {a b : α} (h : a ≤ b) :
+  variation_on_from_to f s a b = (evariation_on f (s ∩ Icc a b)).to_real := if_pos h
+
+lemma variation_on_from_to_eq_of_ge {a b : α} (h : b ≤ a) :
+  variation_on_from_to f s a b = - (evariation_on f (s ∩ Icc b a)).to_real :=
+begin
+  dsimp only [variation_on_from_to],
+  split_ifs with h',
+  { cases le_antisymm h h',
+    rw [Icc_self, evariation_on.subsingleton, ennreal.zero_to_real],
+    simp only [neg_zero],
+    exact λ x hx y hy, hx.2.trans hy.2.symm, },
+  { refl, }
+end
+
+lemma variation_on_from_to_eq_neg_swap (a b : α) :
+  variation_on_from_to f s a b = - variation_on_from_to f s b a :=
```

Comment:
I think you can golf both `variation_on_from_to_nonpos_of_ge` and `variation_on_from_to_eq_of_ge` above using this.